includes/class-payment-information.php

```diff
@@ -148,29 +151,29 @@ public function is_using_saved_payment_method(): bool {
 	 * @return bool True if the payment should be only authorized, false if it should be captured immediately.
 	 */
 	public function is_using_manual_capture(): bool {
-		return $this->manual_capture;
+		return $this->manual_capture->equals( Payment_Capture_Type::MANUAL() );
 	}
 
 	/**
 	 * Payment information constructor.
 	 *
-	 * @param array     $request Associative array containing payment request information.
-	 * @param \WC_Order $order The order object.
-	 * @param bool      $off_session Indicates whether the payment is merchant-initiated (true) or customer-initiated (false).
-	 * @param bool      $manual_capture Indicates whether the payment will be only authorized (true) or captured immediately (false).
+	 * @param array                $request Associative array containing payment request information.
+	 * @param \WC_Order            $order The order object.
+	 * @param Payment_Initiated_By $payment_initiated_by Indicates whether the payment is merchant-initiated or customer-initiated.
+	 * @param Payment_Capture_Type $manual_capture Indicates whether the payment will be only authorized or captured immediately.
 	 *
 	 * @throws \Exception - If no payment method is found in the provided request.
 	 */
 	public static function from_payment_request(
 		array $request,
 		\WC_Order $order = null,
-		bool $off_session = false,
-		bool $manual_capture = false
+		Payment_Initiated_By $payment_initiated_by = null,
+		Payment_Capture_Type $manual_capture = null
 	): Payment_Information {
 		$payment_method = self::get_payment_method_from_request( $request );
 		$token          = self::get_token_from_request( $request );
 
-		return new Payment_Information( $payment_method, $order, $token, $off_session, $manual_capture );
+		return new Payment_Information( $payment_method, $order, $token, $payment_initiated_by ?? Payment_Initiated_By::CUSTOMER(), $manual_capture ?? Payment_Capture_Type::AUTOMATIC() );
```

Comment:
`Payment_Information` already handles null, so no need to check again.

Reply:
Oh that's right, thank you!

Fixed in 8e07e3e351bfadadb80d0d68d05c4a0b24545f1d.